### DISCUSSION TITLE
Corrupted Replicas

### DISCUSSION THREAD

**Discussion by rsoika** [longhorn/longhorn#2291] [Q&A]
Last night I had an issue with one of my PODs running a PostgreSQL DB with a durable longhorn volume. (I am running longhorn 1.1.0)
The problem was, that after a hard hardware reset of two of my 3 nodes the data volume of my POD was corrupted and an error message like the following was shown:

...ext2fs_check_if_mount: Can't check if filesystem is mounted due to missing mtab file ...

I deleted the corrupted replicas and longhorn did rebuild them. After that, the POD started again and postgres is running.

Now my question is: Is it a bad idea to run longhorn on a software raid system? Should I avoid using the software raid?

**Comment by joshimoo** [comment]
@rsoika The above error message seems not related to longhorn, but instead related to the fact that /etc/mtab is not available.
On most distros this should be a symbolic link to `/proc/mounts`

We saw something like this on our share-manager arm tests, on ubuntu 20.04
Which for some reason didn't get created.

What OS/env are your host nodes?

**Reply by rsoika** [reply]
yes, my posted error message was not very precisely.  At the end I answered the question myself. Software raid is not a problem for longhorn. The problem in my case, was the fact that 2 of 3 nodes went down because of a replaced switch in the datacenter of my hoster. So after the nodes started again (15 minutes downtime), longhorn could not decide which replica was the correct one because of missing quorum. This was a extreme situation which should normally not happen. It's important that you do place your nodes in different racks within a datacenter (if possible). This was my main fault not to take care about this.

**Reply by joshimoo** [reply]
@rsoika as long as your 3rd node didn't go down, it should have been used for the replica rebuild.
Are you running with the revision counter disabled by chance?
https://longhorn.io/docs/1.1.0/advanced-resources/deploy/revision_counter/

Did you effectively have a split brain, where the two failed nodes became part of a separate network from node1?
Thank you for reporting we currently don't have tracking on the discussion pages, so messages here might get missed.
If possible it's better to open a [Question] github issue :)
